presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -100,11 +106,28 @@ public void getAllQueryInfo(
             proxyResponse(servletRequest, asyncResponse, xForwardedProto, uriInfo);
             return;
         }
+
+        int limit = firstNonNull(limitFilter, dispatchManager.getMaxQueriesToReturn());
+        if (limit <= 0) {
+            throw new WebApplicationException(Response
+                    .status(BAD_REQUEST)
+                    .type(MediaType.TEXT_PLAIN)
+                    .entity(format("Parameter 'limit' for getAllQueryInfo must be positive. Got %d.", limit))
+                    .build());
+        }
+
         QueryState expectedState = stateFilter == null ? null : QueryState.valueOf(stateFilter.toUpperCase(Locale.ENGLISH));
         ImmutableList.Builder<BasicQueryInfo> builder = new ImmutableList.Builder<>();
+        // TODO: With us having 'limit' now, it would be nice to have 'basic query info' sorted  by some time
+        // (creation time?) in the query list that we get from dispatch manager.
```

Comment:
Can't we retrieve this from #BasicQueryInfo#BasicQueryStats#createTime?

Reply:
Good point.
I didn't notice "#BasicQueryInfo#BasicQueryStats#createTime" :)
Thanks for pointing out.

It is reflected in the task that we need to do this.
We'll do it after.
Now we want to unblock presto cpp, this change won't affect Presto prod in any way.